tests/integration/test_show_answers.py

```diff
@@ -0,0 +1,91 @@
+
+#from xblockutils.studio_editable_test import StudioEditableBaseTest, StudioContainerWithNestedXBlocksBaseTest
+from xblockutils.studio_editable_test import CommonBaseTest
+from web_fragments.fragment import Fragment
+
+
+class ShowAnswerXBlock(ShowAnswerXBlockMixin, XBlock):
+    """
+    A basic ShowAnswer XBlock implementation (for use in tests)
+    """
+    CATEGORY = 'showanswer'
+    STUDIO_LABEL = 'Show Answers'
+
+    color = String(default="red")
+    count = Integer(default=42)
+    comment = String(default="")
+    date = DateTime(default=datetime.datetime(2014, 5, 14, tzinfo=pytz.UTC))
+    editable_fields = ('color', 'count', 'comment', 'date')
+
+    def student_view(self, context):
+        return Fragment()
+
+    def validate_field_data(self, validation, data):
+        """
+        A validation method to check that 'count' is positive and prevent
+        swearing in the 'comment' field.
+        """
+        if data.count < 0:
+            validation.add(ValidationMessage(ValidationMessage.ERROR, u"Count cannot be negative"))
+        if "damn" in data.comment.lower():
+            validation.add(ValidationMessage(ValidationMessage.ERROR, u"No swearing allowed"))
+
+
+@ddt.ddt
+class TestShowAnswerXBlock(SeleniumXBlockTest):
+    """
+    Test the Studio View created for ShowAnswerXBlock
+    """
+
+    def set_up_root_block(self, extra_args=''):
+        self.set_scenario_xml('<showanswer {} />'.format(extra_args))
+        self.go_to_view()
+        return self.load_root_xblock()
+
+    def assert_unchanged(self, block, orig_field_values=None, explicitly_set=False):
+        """
+        Check that all field values on 'block' match with either the value in orig_field_values
+        (if provided) or the default value.
+        If 'explitly_set' is False (default) it asserts that no fields have an explicit value
+        set. If 'explititly_set' is True it expects all fields to be explicitly set.
+        """
+        for field_name in block.editable_fields:
+            expected_value = orig_field_values[field_name] if orig_field_values else block.fields[field_name].default
+            self.assertEqual(getattr(block, field_name), expected_value)
+            self.assertEqual(block.fields[field_name].is_set_on(block), explicitly_set)
+
+    @XBlock.register_temp_plugin(ShowAnswerXBlock, "showanswer")
+    def test_defaults(self):
+        """
+        Assert defaults
+        """
+        block = self.set_up_root_block()
+        self.assertIsNone(block.solution)
+        self.assertEqual(block.showanswer, ShowAnswer.PAST_DUE)
+
+    @XBlock.register_temp_plugin(ShowAnswerXBlock, "showanswer")
+    def test_fields(self):
+        """
+        Assert field overrides in XML
+        """
+        block = self.set_up_root_block('solution="The solution" showanswer="{}"'.format(ShowAnswer.ANSWERED))
+        self.assertEqual(block.solution, 'The solution')
+        self.assertEqual(block.showanswer, ShowAnswer.ANSWERED)
```

Comment:
It doesn't look like anything in this class is being used. What's the reason for adding it in this PR?

Reply:
I copied it from `test_studio_editable`, I'll remove those functions